x/authority/types/policytype_test.go

```diff
@@ -0,0 +1,53 @@
+package types_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/x/authority/types"
+)
+
+func TestPolicyType_Validate(t *testing.T) {
+	tests := []struct {
+		name       string
+		policyType types.PolicyType
+		wantErr    bool
+	}{
+		{
+			name:       "valid groupEmergency",
+			policyType: types.PolicyType_groupEmergency,
+			wantErr:    false,
+		},
+		{
+			name:       "valid groupOperational",
+			policyType: types.PolicyType_groupOperational,
+			wantErr:    false,
+		},
+		{
+			name:       "valid groupAdmin",
+			policyType: types.PolicyType_groupAdmin,
+			wantErr:    false,
+		},
+		{
+			name:       "invalid policy type",
+			policyType: types.PolicyType(20),
```

Comment:
We should check `max+1` for edge case
We can use the len of map object generated by proto